src/mpi/comm/comm_split_type.c

```diff
@@ -144,12 +149,12 @@ static struct shmem_processor_info_table shmem_processor_info[] = {
     {NULL, HWLOC_OBJ_TYPE_MAX}
 };
 
-static int node_split_processor(MPIR_Comm * comm_ptr, int key, hwloc_obj_type_t obj_type,
+static int node_split_processor(MPIR_Comm * comm_ptr, int key, struct info_obj * obj_type,
```

Comment:
What's the reason to change this function signature?  `node_split_processor` and `node_split_io` have different signatures anyway.